src/block_sizes.jl

```diff
@@ -1,15 +1,16 @@
 
+matmul_params(::Val{T}) where {T <: Base.HWReal} = LoopVectorization.matmul_params()
 
-function block_sizes(::Type{T}, _α, _β, R₁, R₂) where {T}
+function block_sizes(::Val{T}, _α, _β, R₁, R₂) where {T}
```

Comment:
Why do we need to wrap the type in a `Val`?

Reply:
To throw an error if not specialized?

Reply:
I don't think we need to, but I think I prefer it as a style.
It should force specialization, just like `Array{T}` always specializes on `T`.